openedx/core/djangoapps/course_views/course_views.py

```diff
@@ -0,0 +1,176 @@
+"""
+Tabs for courseware.
+"""
+from openedx.core.lib.api.plugins import PluginManager
+from xmodule.tabs import CourseTab, link_reverse_func
+
+_ = lambda text: text
+
+
+# Stevedore extension point namespaces
+COURSE_VIEW_TYPE_NAMESPACE = 'openedx.course_view_type'
+
+
+class CourseViewType(object):
+    """
+    Base class of all course view type plugins.
```

Comment:
We should add a more detailed docstring that describes how to enroll new course view types.